scijava/scijava-ops-api/src/main/java/org/scijava/ops/api/package-info.java

```diff
@@ -0,0 +1,46 @@
+ *   <li><b>Structural Equivalence</b></li> Structural Equivalence means that two Ops:
+ *   <ol>
+ *     <li>Are Form Equivalent</li>
+ *     <li>Have the same number of inputs and the same number of outputs</li>
+ *     <li>For each input position, accept form-equivalent inputs</li>
+ *     <li>Return form-equivalent outputs</li>
+ *   </ol>
+ *   <li><b>Result Equivalence</b></li> Result Equivalence means that {@code o1.equals(o2)} for two outputs {@code o1} and {@code o2} from two Ops.
+ * </ol>
+ * Within the Ops API, each type of equivalence is utilized in the following ways:
+ * <ol>
+ *   <li><b>Form Equivalence</b></li> If two Ops are "Form Equivalent", they are defined under the same name.
+ *   <li><b>Structural Equivalence</b></li> If two Ops are "Structural Equivalent", they share a common form-reduced description, searchable using {@link org.scijava.ops.api.OpEnvironment#descriptions(java.lang.String)}. For example, an Op "math.add" that produces an ImgLib2 Img from Img operands, and another Op "math.add" that produces a NumPy ndarray from ndarray operands, will reduce to a single description "math.add" that produces an image from image operands.
+ *   <li><b>Result Equivalence</b></li> If two Ops are "Result Equivalent", then they are considered to produce equivalent values, using the primary language-specific definition of equality (e.g. `Object.equals`, for Java usage).
+ * </ol>
+ *
+ * For example, consider three Ops:
+ * <ol>
+ *   <li> filter.gauss(net.imglib2.img.Img, net.imglib2.type.numeric.real.FloatType) -> net.imglib2.img.Img</li>
+ *   <li> filter.gauss(ij.ImagePlus, java.lang.Double) -> ij.ImagePlus</li>
+ *   <li> filter.gauss(net.imglib2.img.Img, net.imglib2.algorithm.neighborhood.Shape) -> net.imglib2.img.Img</li>
+ * </ol>
+ * Ops 1 and 2 should be considered form-equivalent, as they have the same name, and structural-equivalent, as they both take in an image data structure and a floating point number and return an image data structure, but they are likely not result-equivalent due to the underlying data structures.
+ * <p>
+ * Ops 1 and 3 should also be considered form-equivalent, as they have the same name, but are not structural-equivalent, as one computes its own Shape over which to perform a gaussian blur, while the other takes a predefined shape.
+ * <p>
+ * These definitions of equivalence provide a level of flexibility impossible without the Ops API; structural-equivalence allows us to, for example, define equivalent Ops across programming languages and libraries, and then to create scripts that can run unaltered on data types from each of those languages and libraries. Given the same {@link org.scijava.ops.api.OpEnvironment} and inputs, however, we ensure result-equivalence and therefore reproducability
```

Comment:
This last paragraph should be rewritten. Here is a suggestion:

> These definitions of equivalence provide clarity when articulating the Ops framework's various benefits:
>
> 1. The [ops simplification mechanism](/link/to/simplification/doc) enables structural-equivalent ops to be collapsed into a common simplified form, usable with appropriate data structures across programming languages and libraries.
> 2. ... [out of time to write more today, sorry, but hopefully you see where I'm going with this]

Reply:
@ctrueden I think this is pretty tricky, but I gave it a go in ed64b87b. Let me know what you think.